Sources/AblyChat/RoomLifecycleManager.swift

```diff
@@ -11,31 +17,57 @@ internal protocol RoomLifecycleContributorChannel: Sendable {
 
     var state: ARTRealtimeChannelState { get async }
     var errorReason: ARTErrorInfo? { get async }
+
+    /// Equivalent to subscribing to a `RealtimeChannelProtocol` object’s state changes via its `on(_:)` method. The subscription should use the ``BufferingPolicy.unbounded`` buffering policy.
+    ///
+    /// It is marked as `async` purely to make it easier to write mocks for this method (i.e. to use an actor as a mock).
+    func subscribeToState() async -> Subscription<ARTChannelStateChange>
```

Comment:
Looks like there is nothing else to subscribe to except the `state`, why not just `subscribe` then?

Reply:
You're right that on this protocol there isn't anything else to subscribe to, but since this protocol is quite similar to `RealtimeChannelProtocol`, which has a method called `subscribe` that subscribes to channel messages (i.e. not state), I wanted to avoid confusion by not calling the method `subscribe`.